/*
* @license Apache-2.0
*
* Copyright (c) 2019 The Stdlib Authors.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import incrwmean = require( './index' );

// TESTS //

// The function returns either a number or null...
{
	incrwmean(); // $ExpectType incrwmean_acc
}

// The compiler throws an error if the function is provided arguments...
{
	incrwmean( '5' ); // $ExpectError
	incrwmean( 5 ); // $ExpectError
	incrwmean( true ); // $ExpectError
	incrwmean( false ); // $ExpectError
	incrwmean( null ); // $ExpectError
	incrwmean( undefined ); // $ExpectError
	incrwmean( [] ); // $ExpectError
	incrwmean( {} ); // $ExpectError
	incrwmean( ( x: number ): number => x ); // $ExpectError
}

// The compiler throws an error if the function is provided arguments...
{
	let acc = incrwmean();

	acc( '5' ); // $ExpectError
	acc( 5 ); // $ExpectError
	acc( true ); // $ExpectError
	acc( false ); // $ExpectError
	acc( null ); // $ExpectError
	acc( undefined ); // $ExpectError
	acc( [] ); // $ExpectError
	acc( {} ); // $ExpectError
	acc( ( x: number ): number => x ); // $ExpectError
}
